AWSS3/AWSS3TransferUtility.m

```diff
@@ -1395,6 +1396,9 @@ - (void) retryUpload: (AWSS3TransferUtilityUploadTask *) transferUtilityUploadTa
                 }
             }
             
+            //Save in Database after the file has been created, so that file can be referenced incase upload is paused and needs to be restarted.
```

Comment:
I would put do this if there is no error by adjusting the if statement below. I think that makes sense.

```objc
if (!subTaskCreationError) {
    //Save in Database after the file has been created, so that file can be referenced incase upload is paused and needs to be restarted.
    [AWSS3TransferUtilityDatabaseHelper insertMultiPartUploadRequestSubTaskInDB:transferUtilityMultiPartUploadTask subTask:subTask
databaseQueue:self.databaseQueue];
} else {
    //Abort the request, so the server can clean up any partials.
    [self callAbortMultiPartForUploadTask:transferUtilityMultiPartUploadTask];
    transferUtilityMultiPartUploadTask.status = AWSS3TransferUtilityTransferStatusError;
    //Add it to list of completed Tasks
    [self.completedTaskDictionary setObject:transferUtilityMultiPartUploadTask forKey:transferUtilityMultiPartUploadTask.transferID];
    //Clean up.
    [self cleanupForMultiPartUploadTask:transferUtilityMultiPartUploadTask];
    return [AWSTask taskWithError:subTaskCreationError];
}
```